src/taipy/core/data/data_node.py

```diff
@@ -335,20 +338,39 @@ def _track_edit(self, **options):
         self.last_edit_date = edit.get("timestamp")
         self._edits.append(edit)
 
-    def lock_edit(self):
+    def lock_edit(self, editor_id: Optional[str] = None):
         """Lock the data node modification.
 
         Note:
             The data node can be unlocked with the method `(DataNode.)unlock_edit()^`.
+
+        Parameters:
+            editor_id (Optional[str]): The editor's identifier.
         """
-        self.edit_in_progress = True
+        if editor_id:
+            if self.edit_in_progress and self._editor_id != editor_id and self._editor_expiration_date > datetime.now():
+                raise DataNodeIsBeingEdited(self.id, self._editor_id)
+            self._editor_id = editor_id
+        else:
+            self._editor_id = None
+        self._editor_expiration_date = datetime.now() + timedelta(minutes=self.__EDIT_TIMEOUT)
+        self.edit_in_progress = True  # type: ignore
 
-    def unlock_edit(self):
+    def unlock_edit(self, editor_id: Optional[str] = None):
         """Unlocks the data node modification.
+
         Note:
             The data node can be locked with the method `(DataNode.)lock_edit()^`.
+
+        Parameters:
+            editor_id (Optional[str]): The editor's identifier.
         """
-        self.edit_in_progress = False
+        if editor_id and self._editor_id != editor_id and self._editor_expiration_date > datetime.now():
+            raise DataNodeIsBeingEdited(self.id, self._editor_id)
+        else:
+            self._editor_id = None
+            self._editor_expiration_date = datetime.now()
```

Comment:
There is no expiration date if the editor is None.
I would set the expiration date to None.